kr8s/_config.py

```diff
@@ -182,10 +183,17 @@ class KubeConfig(object):
     def __init__(self, path_or_config: Union[str, Dict]):
         self.path = None
         self._raw = None
+
+        if not path_or_config:
+            raise ValueError("KubeConfig path_or_config is None or empty string.")
         if isinstance(path_or_config, str) or isinstance(path_or_config, pathlib.Path):
             self.path = pathlib.Path(path_or_config).expanduser()
             if not self.path.exists():
                 raise ValueError(f"File {self.path} does not exist")
+            if os.path.isdir(self.path):
```

Comment:
We are already ising `pathlib.Path` so we don't need `os.path`.

```suggestion
            if self.path.is_dir():
```

Reply:
Oops yeah that's better. I had os.path because I was initially making the change in `_auth.py` before realizing the right place was here.